cmd/clusterctl/client/cluster/mover.go

```diff
@@ -742,7 +742,10 @@ func (o *objectMover) createTargetObject(nodeToCreate *node, toProxy Proxy) erro
 	}
 
 	if err := cTo.Create(ctx, obj); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
+		// Kubernetes apiserver responds with a ServerTimeoutError for resources that are created with generateName metadata flag
+		// and the object already exists on the target cluster. The confusing error causes the move command to not be idempotent since
+		// Machine Objects have the generateName flag and so a IsAlreadyExists error check will fail for these objects.
+		if !(apierrors.IsAlreadyExists(err) || (apierrors.IsServerTimeout(err) && obj.GetGenerateName() != "")) {
```

Comment:
Should we opt to remove the generated name if a name exists instead?

Reply:
Yep that would also work! Is this what you would prefer?

Reply:
I think so, we shouldn't rely on the timeout error from Kubernetes; let's form the request in a way that doesn't trigger this strange behavior, then the IsAlreadyExists error should cover all cases.

Reply:
sounds good to me, tested this out locally  to ensure it works as expected and it does.